### PR TITLE
Expanding security assessment facilitator role definition.

### PR DESCRIPTION
**Pull request by IAXES**
- Various edits to appease the `mdl` linter.
- Enforcing some style changes (i.e. common indentation and
  trailing period literals in unordered lists, etc.).
- Using a shell script to auto-generate/align the table of contents
  rendered at the top of the document.

### CODE REVIEW

**Comment by IAXES** [comment]
This was the minimum I could get the linter down to. We'd need a prescribed/pre-created `mdl` linter config file (i.e. that we pull and use locally) if we need 100% of the linter warnings/errors to go away.

```
governance/roles.md:11: MD007 Unordered list indentation
governance/roles.md:82: MD007 Unordered list indentation
governance/roles.md:248: MD013 Line length
governance/roles.md:251: MD013 Line length
```

**Comment by IAXES** [comment]
Amended most recent commit to include signature.

File: governance/roles.md
Comment (ashutosh-narkar) [inline governance/roles.md:222]: Remove extra `the`.

File: governance/roles.md
Comment (ashutosh-narkar) [inline governance/roles.md:224]: Maybe link to https://github.com/cncf/tag-security/issues?q=is%3Aopen+is%3Aissue+label%3Aassessment

File: governance/roles.md
Comment (ashutosh-narkar) [inline governance/roles.md:236]: `at least one additional security assessor`, do you mean `reviewers` ? Also I would imagine we would need atleast 2 reviewers.

File: governance/roles.md
Comment (ashutosh-narkar) [inline governance/roles.md:259]: `assessors` means `reviewers` ?

**Review by ashutosh-narkar — COMMENTED** [review]
@IAXES this looks good. Left a few comments.

File: governance/roles.md
Comment (IAXES) [reply governance/roles.md:224]: Sure.

File: governance/roles.md
Comment (IAXES) [reply governance/roles.md:236]: Agreed. Updating.

File: governance/roles.md
Comment (IAXES) [reply governance/roles.md:259]: Yep :)

Swapping out all occurrences.

**Comment by IAXES** [comment]
Updated as per feedback.

**Comment by ultrasaurus** [comment]
I think it is very important not to repeat information that exists elsewhere, but rather link it.  For example: soft/hard conflicts are presented here: https://github.com/cncf/tag-security/blob/main/assessments/guide/security-reviewer.md  (I think that's a good place for them, though would be open to *moving* rather than copying if the community feels strongly that this information belongs in governance.  If so, consider a small, specific "conflict of interest" doc that can be linked to from both places.)

**Comment by ultrasaurus** [comment]
Note: this is very hard to review because of all the formatting changes included with (maybe?) substantive changes.  Please make clear in the description what expansion of the role you are including, whether this is a proposed change in actual practice or an expanded narrative of what was originally written.  This is critical for anything in `/governance`.

**Comment by IAXES** [comment]
> Note: this is very hard to review because of all the formatting changes included with (maybe?) substantive changes. Please make clear in the description what expansion of the role you are including, whether this is a proposed change in actual practice or an expanded narrative of what was originally written. This is critical for anything in `/governance`.

I'll do a 2-pass approach in the future (i.e. linter + cosmetic changes first round; substantive/major changes in a follow-up). The change is centered around the "Security review facilitator" / "Security Assessment Facilitator" section. The other changes through the document were just style updates and refactoring the table-of-contents at the top.

**Comment by IAXES** [comment]
> I think it is very important not to repeat information that exists elsewhere, but rather link it. For example: soft/hard conflicts are presented here: https://github.com/cncf/tag-security/blob/main/assessments/guide/security-reviewer.md (I think that's a good place for them, though would be open to _moving_ rather than copying if the community feels strongly that this information belongs in governance. If so, consider a small, specific "conflict of interest" doc that can be linked to from both places.)

The existing location (i.e. `security-reviewer`) seems fine, IMHO. I just wanted to have a concrete quote/reference (in this case, copy) of it in the current document. If this was a `rST`/Sphinx doc, for example, I would've just put the example in a separate file and referenced it in both the `governance` and `security-reviewer` docs (i.e. made a macro out of the quote). No objections on my end if we should pull it from the `governance` doc to de-dupe content. :)

**Comment by IAXES** [comment]
@ashutosh-narkar Will need a another :+1: Had to re-base.

File: governance/roles.md
Comment (lumjjb) [inline governance/roles.md:41]: should all be sentence case as per contribution guide

File: governance/roles.md
Comment (lumjjb) [inline governance/roles.md:216]:
```suggestion
sub-section as a facilitator). The facilitator is responsible for
```

File: governance/roles.md
Comment (lumjjb) [inline governance/roles.md:220]: I think the part of this document like @ultrasaurus recommended should be kept brief and pointing to other documents for details. These should be high-level bullet points of what the high level roles and responsibilities are - and the how can be pointed to in the document. A lot of the content that you have here is good, but may be better to move them to the assessment guide page.

An example for illustration of a high level bullet point is:
- Ensure that security reviews follow the assessment process
- Help bootstrap security assessments and reviews
- Determine assessment schedule of reviews, considering TOC requirements
- ...

If needed, you could add a section in the guide on this topic as well.

File: governance/roles.md
Comment (lumjjb) [inline governance/roles.md:259]: Mention here that facilitator will act as a contact point for escalation as well as to do regular check-ins to ensure project is on track.

**Review by lumjjb — CHANGES_REQUESTED** [review]
Thanks for putting this together @IAXES , couple requests for changes!

**Comment by lumjjb** [comment]
Hi @IAXES just checking in if you've had some time to take a look at the comments!

**Comment by IAXES** [comment]
> Hi @IAXES just checking in if you've had some time to take a look at the comments!

Yep: reviewing now. :)

File: governance/roles.md
Comment (IAXES) [reply governance/roles.md:41]: Done. Also updating other heads to follow the same rule.

**Comment by IAXES** [comment]
Updated as per earlier recommendations.

**Comment by lumjjb** [comment]
@IAXES thanks for the update! It seems like there's still quite a few things that are duplicates with existing documents, like the conflict declarations, etc. A lot of that text should be moved out of the `roles.md` document and into the reviewer guide if necessary, roles.md entry should probably be just a paragraph or two.

**Review by anvega — APPROVED** [review]

**Review by anvega — APPROVED** [review]

File: governance/roles.md
Comment (JustinCappos) [inline governance/roles.md:307]: Right now, by rule, the chairs need to do an official sign off.  Are you saying that the facilitator should just make sure that the reviewers have stated their conflict statement or that the facilitator should actually sign off?

Comment (anvega) [reply governance/roles.md:307]: I believe that the intent of the change is for the facilitator to do the signoff. In practice, I see how this proposed change streamlines getting started as is more representative of what happens: Facilitator ensures the reviewers have stated their conflict statement, either takes it at face value, or does quick research to corroborate the statements via LinkedIn and GitHub, and then chases the chairs to comment on the issue with signoff. A chair doing the signoff typically extends the lead time before starting. It has ceremonial value but it's impractical. The issue is public, if the minimum required number of statements is not met or there is no veracity in the statements, it's evident to everyone. The facilitator and the public can do the same check and arithmetic as the chair. The facilitator in its capacity has an interest in high-quality assessments so objectively are impartial.

Comment (JustinCappos) [reply governance/roles.md:307]: Okay, I'm fine with this.  

I do ask that the chairs sign off on this and merge it before I start taking on this myself (as the one in this role should have approval for the rule change). @sublimino @lumjjb @achetal01

Comment (anvega) [reply governance/roles.md:307]: @PushkarJ in lieue of @lumjjb

Comment (lumjjb) [reply governance/roles.md:307]: I do support this, i have no more juice now, but haven't found it to be that valuable since its publicly auditable as stated.

File: governance/roles.md
Comment (lumjjb) [inline governance/roles.md:138]: is this technically possible? I don't believe you can do this in github today? Someone can clarify? @PushkarJ had a bunch of cool github gizmos he implemented through the bot though, so maybe that's possible?

**Review by lumjjb — COMMENTED** [review]
just a minor nit on practicality of permission assignment in github, else LGTM

**Comment by achetal01** [comment]
I m good with the changes...Thank You IAXES for all the hard work in updating the document.

File: governance/roles.md
Comment (anvega) [reply governance/roles.md:138]: It's possible. This was a hindrance when I was the assessed rather than the assessor. We can accomplish so today by declaring ownership of directories or files in CODEOWNERS:  https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#example-of-a-codeowners-file

**Comment by anvega** [comment]
Nudge @sublimino

**Review by JustinCappos — APPROVED** [review]
This looks fine to me.

**Review by mnm678 — APPROVED** [review]
LGTM

**Review by PushkarJ — APPROVED** [review]

**Review by lumjjb — APPROVED** [review]

**Pull request merged**
